transport_test.go

```diff
@@ -125,6 +126,87 @@ func TestGetRequestBodyFromEventCompletelyInvalid(t *testing.T) {
 	}
 }
 
+func TestGetEnvelopeFromBody(t *testing.T) {
+	body := getRequestBodyFromEvent(&Event{
+		Message: "testing",
+	})
+	env := getEnvelopeFromBody(body)
+	envString := env.String()
+	envParts := strings.Split(envString, "\n")
+
+	if len(envParts) != 4 {
+		t.Errorf("\"%s\" should have len of 4, instead has len of %d", envParts, len(envParts))
+	}
+
+	if envParts[3] != "" {
+		t.Errorf("\"%s\" is not empty", envParts[3])
+	}
+
+	var header map[string]interface{}
+	err := json.Unmarshal([]byte(envParts[0]), &header)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sentAtStr := header["sent_at"].(string)
+	if sentAtStr == "" {
+		t.Error("sent_at was not set in header")
+	}
+
+	payload := envParts[2]
+	if payload == "" {
+		t.Error("Item payload is empty")
+	}
+}
+
+func TestGetRequestFromEvent(t *testing.T) {
+	testCases := []struct {
+		testName string
+		// input
+		event *Event
+		// output
+		apiURL string
+	}{
+		{
+			testName: "Sample Event",
+			event:    NewEvent(),
+			apiURL:   "https://host/path/api/42/store/",
+		},
+		{
+			testName: "Transaction",
+			event: func() *Event {
+				event := NewEvent()
+				event.Type = transactionType
+
+				return event
+			}(),
+			apiURL: "https://host/path/api/42/envelope/",
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.testName, func(t *testing.T) {
+			dsn, err := NewDsn("https://key@host/path/42")
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
Please move this out of the loop. The test cases rely on the DSN in their `apiURL` values, and we don't need to re-parse the DSN in every test run.